### PR TITLE
Add support for including paths that are ignored in `.conanignore`

### LINKED ISSUE #17120: [feature] Support inverse match with .conanignore
### What is your suggestion?

As recommended, we store our Conan config in a distinct Git repository. Since we use many automation tools, we end up with numerous config files and other dotfiles in the repository unrelated to Conan. This repo also serves as a template for other project maintainers, who might add more non-Conan related files. `.conanignore` files are helpful for filtering files to be installed. However, they would be more useful if they supported inverse matching, just like `.gitignore` or `.dockerignore` do. This is especially true for the latter, where it's common practice to ignore all files and only explicitly add those required, e.g.:

```ignore
*
!profiles/
!global.conf
!remotes.json
```

I believe this would make maintaining such a config repository much easier. If third-party libraries are an option, there are [gitignore-compatible parsers](https://pypi.org/project/gitignore-parser/) already available for Python.

### Have you read the CONTRIBUTING guide?

- [x] I've read the CONTRIBUTING guide

### PR DESCRIPTION
**Pull request by AbrilRBS**
Changelog: Feature: Add support for including paths that are ignored in `.conanignore`.
Docs: https://github.com/conan-io/docs/pull/3879

Closes: https://github.com/conan-io/conan/issues/17120

### CODE REVIEW

File: conan/internal/api/config/config_installer.py [hunk 1]
```diff
@@ -16,18 +16,30 @@ class _ConanIgnoreMatcher:
...
 
     def matches(self, path):
+        """Returns whether the path should be ignored
+
+        It's ignored if:
+         - The path matches any of the ignored entries
+         - And the path does not match any of the included entries"""
         for ignore_entry in self._ignored_entries:
             if fnmatch.fnmatch(path, ignore_entry):
+                for include_entry in self._included_entries:
```
Comment (memsharded) [inline conan/internal/api/config/config_installer.py]: It is easier and more efficient to do this before iterating the ``ignore_entries`` not inside the loop

File: test/integration/command/config_test.py [hunk 2]
```diff
@@ -89,10 +93,10 @@ def test_config_install_conanignore():
     })
 
     def _assert_config_exists(path):
-        assert os.path.exists(os.path.join(tc.cache_folder, path))
+        assert os.path.exists(os.path.join(tc.cache_folder, path)), f"{path} must exist"
```
Comment (memsharded) [inline test/integration/command/config_test.py]: I think that assert messages in tests in many cases provide little value, but add some visual noise.

**Review by memsharded — COMMENTED** [review]
Looking good, thanks!

File: conan/internal/api/config/config_installer.py (hunk 1 above)
Comment (AbrilRBS) [reply conan/internal/api/config/config_installer.py]: ooops, nice catch! Maybe 9pm is too late to be programming 🤔

**Review by memsharded — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**